src/ngraph/op/fused/softmax_crossentropy.cpp

```diff
@@ -0,0 +1,197 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+#include "ngraph/op/fused/softmax_crossentropy.hpp"
+
+#include "ngraph/builder/make_constant.hpp"
+#include "ngraph/op/constant.hpp"
+#include "ngraph/op/convert.hpp"
+#include "ngraph/op/exp.hpp"
+#include "ngraph/op/log.hpp"
+#include "ngraph/op/max.hpp"
+#include "ngraph/op/multiply.hpp"
+#include "ngraph/op/negative.hpp"
+#include "ngraph/op/not_equal.hpp"
+#include "ngraph/op/one_hot.hpp"
+#include "ngraph/op/reshape.hpp"
+#include "ngraph/op/subtract.hpp"
+#include "ngraph/op/sum.hpp"
+#include "ngraph/op/util/broadcasting.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+constexpr NodeTypeInfo op::SoftmaxCrossEntropy::type_info;
+
+op::SoftmaxCrossEntropy::SoftmaxCrossEntropy(const Output<Node>& arg1,
+                                             const Output<Node>& arg2,
+                                             int64_t reduction_axes,
+                                             bool soft_label,
+                                             int64_t ignore_index)
+    : FusedOp({arg1, arg2})
+    , m_soft_label(soft_label)
+    , m_ignore_index(ignore_index)
+{
+    if (reduction_axes < 0)
+    {
+        m_reduction_axes = arg1.get_shape().size() + reduction_axes;
```

Comment:
This won't work for dynamic shape. You may check this PR: https://github.com/NervanaSystems/ngraph/pull/3708/files